From akpm@osdl.org Mon Aug 14 22:43:29 2006
Message-Id: <200608150543.k7F5hJtw011747@shell0.pdx.osdl.net>
Subject: v4l-dev2: handle __must_check
To: greg@kroah.com
Cc: akpm@osdl.org
From: akpm@osdl.org
Date: Mon, 14 Aug 2006 22:43:19 -0700

From: Andrew Morton <akpm@osdl.org>

We get hundreds of these:

include/media/v4l2-dev.h:348: warning: ignoring return value of 'class_device_create_file', declared with attribute warn_unused_result

Handle it, and propagate the __must_check back a level.

Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 include/media/v4l2-dev.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/include/media/v4l2-dev.h
+++ gregkh-2.6/include/media/v4l2-dev.h
@@ -341,7 +341,7 @@ extern int video_usercopy(struct inode *
 extern struct video_device* video_devdata(struct file*);
 
 #define to_video_device(cd) container_of(cd, struct video_device, class_dev)
-static inline int
+static inline int __must_check
 video_device_create_file(struct video_device *vfd,
 			 struct class_device_attribute *attr)
 {
